fabric/src/test/java/TestStyledText.java

```diff
@@ -530,8 +530,12 @@ public void styledText_shouldProduceCorrectComponent() {
 
         StyledText styledText = StyledText.fromComponent(component);
 
+        // The reconstructed componenet differs in that StyledText always adds components as a sibling to an empty
+        // component
         Assertions.assertEquals(
-                component, styledText.getComponent(), "StyledText.getComponent() returned an unexpected value.");
+                component.toFlatList(),
+                styledText.getComponent().toFlatList(),
+                "StyledText.getComponent() returned an unexpected value.");
```

Comment:
This is not fully ideal, but at least we produce a more correct component (at least, it behaves like the original one, even if the structures differ).